### PR TITLE
Update for GE 4.2 compatibility

### PR DESCRIPTION
**Pull request by dfev77**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build related changes
- [ ] Documentation
- [ ] Other: 

## What is the current behavior?
Cannot publish gerrit changes

Issue Number: #80 

## What is the new behavior?
Can publish gerrit changes

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

Not compatible with GE < 4.2 due to breaking changes in GE

### CODE REVIEW

**Review by RussKie — APPROVED** [review]

**Review by pmiossec — APPROVED** [review]

**Comment by dfev77** [comment]
FYI: There were 2 breaking changes that caused the issue
* removal of AppSettings.GitBinDir -> alternative: add the new property and made the old one Obsolete(true)
* adding an optional argument to IGitModule.GetSelectedBranch -> alternative: add an overload and have the original method call the new one with the default argument value.

The alternatives mentioned above for each case would have had maintained binary compatibility and it would had allowed the plugin to be compatible with the new GE version

**Review by gerhardol — APPROVED** [review]

**Comment by RussKie** [comment]
> FYI: There were 2 breaking changes that caused the issue
> 
> * removal of AppSettings.GitBinDir -> alternative: add the new property and made the old one Obsolete(true)
> * adding an optional argument to IGitModule.GetSelectedBranch -> alternative: add an overload and have the original method call the new one with the default argument value.
> 
> The alternatives mentioned above for each case would have had maintained binary compatibility and it would had allowed the plugin to be compatible with the new GE version

Yes, and we need to be more aware of this. 
With https://github.com/gitextensions/gitextensions/pull/11269 we'll be in a much better position to finally move the shared abstractions to GitExtensions.Extensibility library, and then bolt [Roslyn API analysers](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) on to help us either documenting breaking changes publicly or avoiding those, if possible.

**Comment by mycroes** [comment]
> Yes, and we need to be more aware of this. With [gitextensions/gitextensions#11269](https://github.com/gitextensions/gitextensions/pull/11269) we'll be in a much better position to finally move the shared abstractions to GitExtensions.Extensibility library, and then bolt [Roslyn API analysers](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md) on to help us either documenting breaking changes publicly or avoiding those, if possible.

That's so good to hear. We get struck by the incompatible Gerrit plugin on every breaking change. I usually just wait until it's fixed and push through commandline in the meantime, but the rest of the team kinda relies on a working Gerrit plugin, so it has become kind of a running joke who will update GitExtensions first. This one has taken a while to get noticed/fixed though, so improving this situation is very much appreciated!

File: src/GitExtensions.GerritPlugin/GitExtensions.GerritPlugin.nuspec [hunk 1]
```diff
@@ -11,11 +11,11 @@
     <icon>Resources\IconGerrit.png</icon>
     <dependencies>
       <group>
-        <dependency id="GitExtensions.Extensibility" version="[0.3.0, 0.4.0)" />
+        <dependency id="GitExtensions.Extensibility" version="[0.3.1, 0.4.0)" />
       </group>
       <!-- To fix Warning NU5128 Add a dependency group for net6.0-windows7.0 to the nuspec -->
       <group targetFramework="net6.0-windows7.0">
-        <dependency id="GitExtensions.Extensibility" version="[0.3.0, 0.4.0)" />
+        <dependency id="GitExtensions.Extensibility" version="[0.3.1, 0.4.0)" />
```
Comment (lekhmanrus) [inline src/GitExtensions.GerritPlugin/GitExtensions.GerritPlugin.nuspec:18]: Should it be still 0.3.0 for the existing GitExtensions v4.2? Looks like the newly released version of the plugin (2.0.7) can't be discovered by the GE Plugin Manager for some reason.

Comment (dfev77) [reply src/GitExtensions.GerritPlugin/GitExtensions.GerritPlugin.nuspec:18]: Yes. I've only changed them as I was struggling to understand how to get the 4.2 deps.
It took me a while to just read the msbuild targets in order to clarify this.

**Pull request merged**